src/State.hs

```diff
@@ -1286,6 +1286,31 @@ deleteMessage new = do
   cId <- use csCurrentChannelId
   when (postChannelId new == cId) updateViewed
 
+maybePostUsername :: ChatState -> Post -> T.Text
+maybePostUsername st p = case do { uId <- postUserId p; name <- usernameForUserId uId st; return name } of
+    Nothing -> T.empty
+    Just x -> x
+
+maybeNotify :: [PostToAdd] -> MH ()
```

Comment:
The `maybeNotify` function is written to work on lists of `PostToAdd`, but it should be written more simply to just run on a single `PostToAdd` and then the call site in `postOp` would instead say `mapM_ maybeNotify p`.